oak-store-composite/src/main/java/org/apache/jackrabbit/oak/composite/checks/NodeTypeDefinitionNodeStoreChecker.java

```diff
@@ -75,6 +75,11 @@ public void onConstraintViolation(String path, List<String> typeNames, int code,
         NodeBuilder rootBuilder = new ReadOnlyBuilder(root); // prevent accidental changes
         
         String primary = root.getName(JCR_PRIMARYTYPE);
+        // workaround until https://issues.apache.org/jira/browse/OAK-9863 is fixed
```

Comment:
@rombert Do you plan on working OAK-9863 or are you fine with this temporary workaround for now?

Reply:
Let's go with this one for now.